src/WingetCreateCLI/Commands/UpdateCommand.cs

```diff
@@ -199,44 +200,33 @@ public async Task<Manifests> DeserializeExistingManifestsAndUpdate(List<string>
 
-            if (!PackageParser.UpdateInstallerNodesAsync(
-                installerManifest,
-                this.InstallerUrls,
-                packageFiles,
-                out bool installerMismatch,
-                out List<WingetCreateCore.Models.Installer.Installer> unmatchedInstallers,
-                out List<WingetCreateCore.Models.Installer.Installer> multipleMatchedInstallers,
-                out List<PackageParser.DetectedArch> detectedArchOfInstallers))
-            {
-                DisplayMismatchedArchitectures(detectedArchOfInstallers);
-
-                if (installerMismatch)
-                {
-                    Logger.ErrorLocalized(nameof(Resources.NewInstallerUrlMustMatchExisting_Message));
-
-                    if (unmatchedInstallers.Any())
-                    {
-                        Logger.ErrorLocalized(nameof(Resources.InstallerMatchFailedError_Message));
-                        Console.WriteLine();
-                        unmatchedInstallers.ForEach(i => Logger.ErrorLocalized(nameof(Resources.InstallerDetectedFromUrl_Message), i.Architecture, i.InstallerType, i.InstallerUrl));
-                    }
+            List<PackageParser.DetectedArch> detectedArchOfInstallers;
 
-                    if (multipleMatchedInstallers.Any())
-                    {
-                        Logger.ErrorLocalized(nameof(Resources.MultipleMatchingInstallerNodes_Error));
-                        Console.WriteLine();
-                        multipleMatchedInstallers.ForEach(i =>
-                        {
-                            Logger.ErrorLocalized(nameof(Resources.InstallerDetectedFromUrl_Message), i.Architecture, i.InstallerType, i.InstallerUrl);
-                            Console.WriteLine();
-                        });
-                    }
-                }
-                else
-                {
-                    Logger.ErrorLocalized(nameof(Resources.PackageParsing_Error));
-                }
+            try
+            {
+                PackageParser.UpdateInstallerNodesAsync(
+                    installerManifest,
+                    this.InstallerUrls,
+                    packageFiles,
+                    out detectedArchOfInstallers);
+            }
+            catch (InvalidOperationException)
+            {
+                Logger.ErrorLocalized(nameof(Resources.InstallerCountMustMatch_Error));
+                return null;
+            }
+            catch (ParsePackageException parsePackageException)
+            {
+                parsePackageException.ParseFailedInstallerUrls.ForEach(i => Logger.ErrorLocalized(nameof(Resources.PackageParsing_Error), i));
+                return null;
+            }
+            catch (InstallerMatchException installerMatchException)
+            {
+                Logger.ErrorLocalized(nameof(Resources.NewInstallerUrlMustMatchExisting_Message));
+                Logger.ErrorLocalized(nameof(Resources.InstallersFailedToMatch_Message));

```

Comment:
Are these ever used independently? If not, consider merging the resources as it adds more context for better localization.

Reply:
They were before, but not anymore. I merged the resources into a single resource string so that context is more clear.